test: configure the windows test environment

compnerd opened the pull request:
This enables lit to at least start running on Windows.  There is more
work necessary to get the tests actually passing on Windows.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
CC: @Rostepher @bob-wilson @jrose-apple @troughton @hughbe @AndrewSB @gottesmm

compnerd commented:
@swift-ci please test

jrose-apple commented on test/lit.cfg:
```diff
@@ -158,8 +158,16 @@ if platform.system() == 'Darwin':
         os.environ.get('TOOLCHAINS', config.darwin_xcrun_toolchain)
 
+
+kIsWindows = platform.system() == 'Windows'
```
Nitpick: please match the naming conventions for all the other variables in the file (snake_case, no leading "k").

compnerd replied:
This is a bit more complicated: this is also the name of a variable in the rest of lit (where I stole it from).  It is in the core lit infrastructure ... so I'm kinda torn (my personal preference is the snake case).

jrose-apple replied:
Ah. Uh…we're *setting* a variable in the rest of lit? Is it expecting us to do that?

Basically, if it's what Clang does then okay (although it may be worth leaving a comment).

jrose-apple reviewed (commented):
Looks good to me!

jrose-apple commented:
Ah, is it worth bothering to make sure this doesn't conflict with Cygwin? I don't know if anyone's actively using that, though.

compnerd commented:
IIRC, @huonw was the one working on cygwin?  I don't know if the tests ever passed in that environment in the first place.

huonw commented:
Not me, sorry!

troughton commented:
I think @tinysun212 was working on the Cygwin port.

tinysun212 commented:
It is difficult to test on the Cygwin in my current situation.
I think this PR is not likely to affect to Cygwin because the added codes are enclosed by `elif run_os in ['windows-msvc']:`.

compnerd commented:
@swift-ci please test

AndrewSB commented:
This is awesome 👍

compnerd commented:
Okay, I've not had to tweak this in quite some time.  I think that this is good from the support side, the tests need adjustment still.

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please tet

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test and merge

compnerd commented:
@swift-ci please test and merge

compnerd commented:
Please test with following PR:
https://github.com/apple/swift-llvm/pull/81

@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please smoke test

compnerd committed:
test: configure the windows test environment

This enables lit to at least start running on Windows.  There is more
work necessary to get the tests actually passing on Windows.

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please test and merge
